internal/download/get.go

```diff
@@ -75,14 +79,25 @@ func httpGetWithProgress(url string, progress *progress.Progress) ([]byte, *fail
 
 	bar := progress.AddByteProgressBar(int64(total))
 
+	// Ensure bar is always closed (especially for retries)
+	defer func() {
+		if !bar.Completed() {
+			bar.Abort(true)
+		}
+	}()
+
 	src := resp.Body
 	var dst bytes.Buffer
 
 	src = bar.ProxyReader(resp.Body)
 
 	_, err = io.Copy(&dst, src)
 	if err != nil {
-		return nil, failures.FailInput.Wrap(err)
+		logging.Debug("Reading body failed: %s", err)
+		if attempt < 3 {
+			return httpGetWithProgressRetry(url, progress, attempt + 1, retries)
+		}
+		return nil, failures.FailNetwork.Wrap(err)
 	}
 
 	if !bar.Completed() {
```

Comment:
With this same logic being contained in a deferred function, it should be safe to remove this.

Reply:
The thing under the conditional is different. This ones increasing, the other one is aborting.